rmw_zenoh_cpp/src/rmw_subscriber.cpp

```diff
@@ -180,17 +222,58 @@ rmw_destroy_subscription(rmw_node_t * node, rmw_subscription_t * subscription)
                                    eclipse_zenoh_identifier,
                                    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
 
-  RCUTILS_LOG_DEBUG_NAMED("rmw_zenoh_cpp", "[rmw_destroy_subscription] %s",
-                          subscription->topic_name);
+  // OBTAIN SUBSCRIPTION MEMBERS ===============================================
+  auto subscription_data = static_cast<rmw_subscription_data_t *>(subscription->data);
+
+  RCUTILS_LOG_DEBUG_NAMED("rmw_zenoh_cpp",
+                          "[rmw_destroy_subscription] %s (ID: %ld)",
+                          subscription->topic_name,
+                          subscription_data->subscription_id_);
 
   // OBTAIN ALLOCATOR ==========================================================
   rcutils_allocator_t * allocator = &node->context->options.allocator;
 
-  // CLEANUP ===================================================================
-  zn_undeclare_subscriber(
-      static_cast<rmw_subscription_data_t *>(subscription->data)->zn_subscriber_);
+  // DELETE SUBSCRIPTION DATA IN TOPIC MAP =========================================================
+  std::string key(subscription->topic_name);
+  auto map_iter = rmw_subscription_data_t::zn_topic_to_sub_data.find(key);
+
+  if (map_iter == rmw_subscription_data_t::zn_topic_to_sub_data.end()) {
+    RCUTILS_LOG_WARN_NAMED("rmw_zenoh_cpp",
+                           "subscription not found in Zenoh topic to subscription data map! %s",
+                           subscription->topic_name);
+  }
+
+  // Delete the subscription data pointer in the Zenoh topic to subscription data map
+  for (auto it = map_iter->second.begin(); it != map_iter->second.end(); ++it) {
```

Comment:
Should this be inside an `else` block for the `if` from line 240?

Reply:
Nice catch: ee6d6d282354e63494d82aba042579868497b782